src/diff/patch.js

```diff
@@ -28,19 +29,14 @@ import { Fragment } from '../create-element';
  * @param {import('../internal').Internal} internal The Internal node to patch
  * @param {import('../internal').CommitQueue} commitQueue List of components
  * which have callbacks to invoke in commitRoot
- * @param {import('../internal').PreactNode} startDom
  */
-export function patch(parentDom, newVNode, internal, commitQueue, startDom) {
+export function patch(parentDom, newVNode, internal, commitQueue) {
 	let dom = internal._dom;
 	let flags = internal.flags;
 
-	if (flags & TYPE_TEXT) {
-		if (newVNode !== internal.props) {
-			dom.data = newVNode;
-			internal.props = newVNode;
-		}
-
-		return dom.nextSibling;
+	if (flags & TYPE_TEXT && newVNode !== internal.props) {
+		dom.data = newVNode;
+		internal.props = newVNode;
 	}
```

Comment:
This needs to always bail out of patch() for Text nodes, even when they're not updated:


```suggestion
	if (flags & TYPE_TEXT) {
		if (newVNode !== internal.props) {
			dom.data = newVNode;
			internal.props = newVNode;
		}

		return;
	}
```